src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7347,15 +7347,22 @@ private StatementSyntax ParseEmbeddedStatement(bool complexCheck)
                 statement = SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken));
             }
 
-            // An "embedded" statement is simply a statement that is not a labelled
-            // statement or a declaration statement.
             switch (statement.Kind)
             {
+                // An "embedded" statement is simply a statement that is not a labelled
+                // statement or a declaration statement.
                 case SyntaxKind.LabeledStatement:
                 case SyntaxKind.LocalDeclarationStatement:
                 case SyntaxKind.LocalFunctionStatement:
                     statement = this.AddError(statement, ErrorCode.ERR_BadEmbeddedStmt);
                     break;
+                // Expression statements as embedded statements should be followed by semicolon.
+                case SyntaxKind.ExpressionStatement:
+                    if (((ExpressionStatementSyntax)statement).SemicolonToken.IsMissing)
+                    {
+                        statement = this.AddError(statement, ErrorCode.ERR_SemicolonExpected);
```

Comment:
Thank you for comments! 

Synced with AlekseyTs and agreed to check if the problem should be addressed on the IncrementalParser side. I added tests with the new iteration: it seems that IncrementalParser is fine (or otherwise my tests are incorrect).

I added a condition making new error possible in scripting scenarios only. It seems to work but looks fragile. I will sync with Gafter and AlekseyTs tomorrow.